libcob/fileio.c

```diff
@@ -3414,17 +3404,36 @@ cob_file_save_status (cob_file *f, cob_field *fnstatus, const int status)
 			break;
 		}
 	} else {
+		f->file_status[0] = (unsigned char)COB_I2D (status / 10);
+		f->file_status[1] = (unsigned char)COB_I2D (status % 10);
+	}
+	if (fnstatus) {
+		memcpy (fnstatus->data, f->file_status, (size_t)2);
+	}
+	if (f->file_status[0] == '0'
+	 && (!delete_file_status || status == 0)) {
+		/* EOP is non-fatal therefore 00 status but needs exception */
+		if (eop_status == 0) {
+			file_globptr->cob_exception_code = 0;
+		} else {
+#if 0 /* correct thing to do, but then also needs to have codegen adjusted
+         --> module-incompatibility --> 4.x */
+			cob_set_exception (eop_status);
+#else
+			cob_set_exception (COB_EC_I_O_EOP);
+#endif
```

Comment:
either inspect or move to TODO entry

Reply:
I don't know what codegen adjustments would be needed (though boldly replacing 0 by 1 doesn't seem to break the testsuite...).

Reply:
Added to the TODO list